### PR TITLE
[NET-689] Restrict Button Access

### PR DESCRIPTION
**Pull request by doblanken**
[NET-689](https://nethinks.atlassian.net/browse/NET-689)

### CODE REVIEW

File: app/src/app/auth/directives/permission-link.directive.ts [hunk 1]
```diff
@@ -19,6 +19,9 @@
 
 import { Directive, ElementRef, Input, TemplateRef, ViewContainerRef } from '@angular/core';
 import { PermissionService } from '../services/permission.service';
+import {CmdbType} from '../../framework/models/cmdb-type';
```
Comment (markheumueller) [inline app/src/app/auth/directives/permission-link.directive.ts:22]: Minor cosmetic fault - The space between the characters is not correct.

File: app/src/app/auth/services/acl-permission.service.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
+import {CmdbType} from '../../framework/models/cmdb-type';
```
Comment (markheumueller) [inline app/src/app/auth/services/acl-permission.service.ts:2]: The blank spaces are not followed.

File: app/src/app/auth/services/acl-permission.service.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Injectable } from '@angular/core';
```
Comment (markheumueller) [inline app/src/app/auth/services/acl-permission.service.ts:1]: You have forgotten the file header here.

File: app/src/app/framework/models/cmdb-render.ts [hunk 4]
```diff
@@ -46,6 +46,7 @@ export class RenderResult {
     active: boolean;
     version: string;
     icon: string;
+    acl: any;
```
Comment (markheumueller) [inline app/src/app/framework/models/cmdb-render.ts:49]: Under `app/acl/acl.types.ts` is a class called **AccessControlList**.

File: app/src/app/framework/object/components/object-actions/object-actions.component.ts [hunk 5]
```diff
@@ -5,6 +5,7 @@ import { ApiCallService } from '../../../../services/api-call.service';
 import { RenderResult } from '../../../models/cmdb-render';
 import {NgbModalRef} from '@ng-bootstrap/ng-bootstrap';
 import {SidebarService} from '../../../../layout/services/sidebar.service';
+import {AccessControlList} from '../../../../acl/acl.types';
```
Comment (markheumueller) [inline app/src/app/framework/object/components/object-actions/object-actions.component.ts:8]: Fileheader + Spacing

File: app/src/app/auth/directives/permission-link.directive.ts [hunk 6]
```diff
@@ -42,30 +52,44 @@ export class PermissionLinkDirective {
...
+  }
+
+  @Input('permissionLinkAcl') set permissionLinkType(acl: AccessControlList) {
+    this.acl = acl;
+    this.updateView();
   }
 
+
+
   @Input() routerLink: string = undefined;
```
Comment (markheumueller) [inline app/src/app/auth/directives/permission-link.directive.ts:66]: You do not have to initialize with undefined - it´s optional.

File: app/src/app/framework/object/components/object-table-actions/object-table-actions.component.ts [hunk 7]
```diff
@@ -22,6 +22,7 @@ import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
 import { ObjectPreviewModalComponent } from '../../modals/object-preview-modal/object-preview-modal.component';
 import { ObjectDeleteModalComponent } from '../../modals/object-delete-modal/object-delete-modal.component';
 import { ReplaySubject } from 'rxjs';
+import {AccessControlList} from '../../../../acl/acl.types';
```
Comment (markheumueller) [inline app/src/app/framework/object/components/object-table-actions/object-table-actions.component.ts:25]: Spaces

File: app/src/app/framework/object/components/object-table-head/object-table-head.component.ts [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import {Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output} from '@angular/core';
```
Comment (markheumueller) [inline app/src/app/framework/object/components/object-table-head/object-table-head.component.ts:1]: Spaces + Header

File: app/src/app/framework/object/object-view/object-view.component.ts [hunk 9]
```diff
@@ -51,7 +51,7 @@ export class ObjectViewComponent implements OnInit, OnDestroy {
   }
 
   public ngOnInit(): void {
-    this.objectViewSubject.asObservable().pipe(takeUntil(this.unsubscribe)).subscribe((result: RenderResult) => {
+    this.objectViewSubject.asObservable().subscribe((result) => {
```
Comment (markheumueller) [inline app/src/app/framework/object/object-view/object-view.component.ts:54]: Why did you delete the unsubscribe?

**Review by markheumueller — CHANGES_REQUESTED** [review]
Personally, I don't like the microsyntax approach from a purely visual point of view, because for me it is a break in reading quality. But I guess that you left it for compatibility reasons with the old syntax. Therefore it is ok.

File: app/src/app/framework/object/object-view/object-view.component.ts (hunk 9 above)
Comment (doblanken) [reply app/src/app/framework/object/object-view/object-view.component.ts:54]: Suggestion of @johndgrey

**Review by markheumueller — APPROVED** [review]
LGTM
